Tasks/NuGetInstaller/VsoNuGetHelper.ps1

```diff
@@ -121,6 +121,19 @@ function SetCredentialsNuGetConfigAndSaveTemp
         [uri]$targetFeed
     )
 
+    #check that the config has a configuration section
+    if($nugetConfig.configuration -eq $null)
+    {
+        if($nugetConfig.packages -ne $null)
+        {
+            throw (Get-LocalizedString -Key "An invalid NuGet.config was passed to the NuGet Installer step. Confirm you did not add packages.config where NuGet.Config was expected")
```

Comment:
The first sentence is fine by me. Second sentence: Check the settings for this build step and confirm you selected NuGet.config rather than packages.config.